frontend/src/metabase-lib/lib/metadata/Metric.ts

```diff
@@ -1,35 +1,46 @@
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-nocheck
 import Base from "./Base";
+import { MetricId } from "metabase-types/types/Metric";
+import {
+  StructuredQuery as StructuredQueryType,
+  MetricAgg,
+} from "metabase-types/types/Query";
+import Table from "./Table";
+import Database from "./Database";
+import Aggregation from "../queries/structured/Aggregation";
+import StructuredQuery from "../queries/StructuredQuery";
+
 /**
  * @typedef { import("./metadata").Aggregation } Aggregation
  */
 
 /**
  * Wrapper class for a metric. Belongs to a {@link Database} and possibly a {@link Table}
  */
-
 export default class Metric extends Base {
+  id!: MetricId;
```

Comment:
Adding the `!` tells TypeScript we are guaranteeing this property exists on `Metric` since it can't figure out that `Base` adds properties to the instance.